### ISSUE TITLE
Define 'ip_address = "" ' parameter in "fixed_ip"  for openstack_networking_port_v2 ,but the ip will be assigned a value.

### ISSUE THREAD

**Issue by xudehua** [terraform-providers/terraform-provider-openstack#429]
Hi there,

Terraform v0.11.7
+ provider.openstack v1.9.0

### Affected Resource(s)
Please list the resources as a list, for example:
- openstack_compute_instance_v2
- openstack_networking_port_v2

According the terraform docs https://www.terraform.io/docs/providers/openstack/r/networking_port_v2.html

The fixed_ip block supports:
    subnet_id - (Required) Subnet in which to allocate IP address for this port.
    ip_address - (Optional) IP address desired in the subnet for this port. If you don't specify ip_address, an available IP address from the specified subnet will be allocated to this port. **This field will not be populated if it is left blank.** To retrieve the assigned IP address, use the all_fixed_ips attribute.

    I create port and instance
resource "openstack_networking_port_v2" "port_net-03_v4" {
    name = "test"
    network_id = "${var.uuid}"
    admin_state_up = "true"
    security_group_ids = [ "${var.security_group_id}" ]
    fixed_ip {
        "subnet_id" = "${var.sub_uuid}"
        "ip_address" = ""
    }
}

 Output
   I just want to create a port in the instance ,no IP required. But after creating instance , the instance's port had been assigned IP value( may be by DHCP).

Thanks

**Comment by ozerovandrei** [comment]
@xudehua Hello.

This is expected behaviour according to [OpenStack Neutron Documentation](https://developer.openstack.org/api-ref/network/v2/#create-port):

> If you specify only a subnet ID, OpenStack Networking allocates an available IP from that subnet to the port.

And we have the following note in the [Terraform Provider Documentation](https://www.terraform.io/docs/providers/openstack/r/networking_port_v2.html):

> If you don't specify ip_address, an available IP address from the specified subnet will be allocated to this port.

Setting this value to blank is the same as skipping this flag. 
Let me know if you have any questions.

**Comment by xudehua** [comment]
@ozerovandrei  Hi,

Thank you for you answer.
By terraform *.tf file,I created a port from the specified subnet and attached to one VM on openstack platformthe .I hope the port's IP address is blank when the VM was deployed.
Maybe I misunderstood the note “**This field will not be populated if it is left blank.**”. I thought that  the port's IP address would not be assigned when the "ip_address" was set blank in the fixed_ip.

Thanks again

Edward

**Comment by jtopjian** [comment]
> I thought that the port's IP address would not be assigned when the "ip_address" was set blank in the fixed_ip.

That's a good point. We will update our documentation to clarify "if it is left blank or omitted".

As far as creating a port with no IP address, you might have to disable DHCP on the subnet itself.

**Comment by xudehua** [comment]
@jtopjian 

Can you describe it in more detail. I disable DHCP,but  the VM was still assigned IP address.

Another, Did the Terraform support the paramete "--no-fixed-ip" in the command " openstack port create **--no-fixed-ip**"? 

Edward

**Comment by xudehua** [comment]
--no-fixed-ip         No IP or subnet for this port.

**Comment by xudehua** [comment]
openstack port create --help

--no-fixed-ip         No IP or subnet for this port.

**Comment by jtopjian** [comment]
I don't think it currently supports it, but we _might_ be able to get this to work. Here's the API request when using `--no-fixed-ip`:

```json
{"port": {"network_id": "foo", "fixed_ips": [], "name": "jttest", "admin_state_up": true}
```

The upstream Gophercloud library might allow us to pass in `[]` for the value of `fixed_ips`. I'll test this out when I have a moment.

**Comment by ozerovandrei** [comment]
@jtopjian I guess that `[]` actually _might_ work: [db_base_plugin_v2.py#L1322](https://github.com/openstack/neutron/blob/stable/rocky/neutron/db/db_base_plugin_v2.py#L1322)

**Comment by ozerovandrei** [comment]
Implemented in #433 

@xudehua this feature will be available in `1.10.0` release.
